pkg/controller/machinepool/nutanixactuator.go

```diff
@@ -0,0 +1,118 @@
+package machinepool
+
+import (
+	"fmt"
+	machinev1 "github.com/openshift/api/machine/v1"
+	machineapi "github.com/openshift/api/machine/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	installvsphere "github.com/openshift/installer/pkg/asset/machines/vsphere"
+	installertypes "github.com/openshift/installer/pkg/types"
+	installertypesnutanix "github.com/openshift/installer/pkg/types/nutanix"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NutanixActuator encapsulates the pieces necessary to be able to generate
+// a list of MachineSets to sync to the remote cluster
+type NutanixActuator struct {
+	client client.Client
+	logger log.FieldLogger
+	scheme *runtime.Scheme
+	//nutanixClient *nutanixclient.Client
+}
+
+var _ Actuator = &NutanixActuator{}
+
+// NewNutanixActuator is the constructor for building a NutanixActuator
+func NewNutanixActuator(client client.Client, scheme *runtime.Scheme, logger log.FieldLogger) (*NutanixActuator, error) {
+	//nutanixClient, err := nutanixclient.NewClient()
+	//if err != nil {
+	//	return nil, errors.Wrap(err, "failed to create Nutanix session")
+	//}
+
+	actuator := &NutanixActuator{
+		client: client,
+		scheme: scheme,
+		//nutanixClient: nutanixClient,
```

Comment:
Let's talk about whether you're actually going to need this. In general, the major cloud bit -- managing VMs -- is done way down the line by MAO, using the cloud creds on the spoke, operating on the MachineSets we generate. Exceptions exist: for example, in [GCP](https://github.com/openshift/hive/blob/e4b80d0c1e6a97443a18594ff42fbf85651a6171/pkg/controller/machinepool/gcpactuator.go#L275) and [IBMCloud](https://github.com/openshift/hive/blob/cb1d134550919670e4307c671e52fb923d0d1925/pkg/controller/machinepool/ibmcloudactuator.go#L82) we need to discover zone names. But by default I would say YAGNI.